core/src/main/java/org/jdbi/v3/core/mapper/OptionalMapperFactory.java

```diff
@@ -38,28 +42,59 @@
  * </ul>
  */
 class OptionalMapperFactory implements ColumnMapperFactory {
-    private final Map<Class<?>, ColumnMapper<?>> mappers = new IdentityHashMap<>();
-
-    OptionalMapperFactory() {
-        mappers.put(OptionalInt.class, optionalMapper(ResultSet::getInt, OptionalInt::empty, OptionalInt::of));
-        mappers.put(OptionalLong.class, optionalMapper(ResultSet::getLong, OptionalLong::empty, OptionalLong::of));
-        mappers.put(OptionalDouble.class, optionalMapper(ResultSet::getDouble, OptionalDouble::empty, OptionalDouble::of));
-    }
+    private static final Map<Class<?>, OptionalStrategy> STRATEGIES =
+            Collections.unmodifiableMap(Arrays.stream(OptionalStrategy.values())
+                .collect(Collectors.toMap(
+                        OptionalStrategy::optClass,
+                        Function.identity(),
+                        (u, v) -> {
```

Comment:
I think `collect(collectingAndThen(toMap(OptionalStrategy::optClass, identity()), IdentityHashMap::new))` is more suitable here, so you don't have to redundantly specify the throwingMerger

Reply:
It's a little nicer to not have to specify the collector arg, but it does result in a redundant map copy, which is a little sad.

Reply:
I'm actually not sure why this uses `IdentityHashMap` in the first place.  `Class` doesn't even override `equals` so there shouldn't be any behavioral differences.